### PR TITLE
INT-1513 Feature/accept oauth tokens

### PR DESCRIPTION
**Pull request by TomPendergrass**

### CODE REVIEW

File: src/modules/bbcore.auth.js [hunk 1]
```diff
@@ -163,6 +163,16 @@ BBCore.prototype.validateAccessToken = function (onSuccess) {
...
+/**
+ * Detects if accessToken provided is an api_key or an accessToken
+ */
+BBCore.prototype.isAccessToken = function (value) {
+    if(typeof value === 'string' && value.length > 32 && value.indexOf('.') > -1) {
+        return true;
+    }
+
+    return false;
+}
```
Comment (BenRussert) [inline src/modules/bbcore.auth.js:171]: You could go:

```js
return (
    typeof value === "string" && value.length > 32 && value.indexOf(".") > -1
  );
```

File: spec/BBCoreSpec.js [hunk 2]
```diff
@@ -396,6 +396,17 @@ describe("BBCore.auth", function() {
...
+        var key = 'a1a1a1a1-1aaa-11a1-1aa1-a11a11aa1111';
+        var token = 'aaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAA.aaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAaaAaaAAaaaAAAaAAaaAaaaAAAah!';
+        var isKey = bbCore.isAccessToken(key);
+        expect(isKey).toBe(false);
+
+        var isToken = bbCore.isAccessToken(token);
+        expect(isToken).toBe(true);
+       
+    });
+
```
Comment (BenRussert) [inline spec/BBCoreSpec.js:409]: 💯

File: README.md [hunk 3]
```diff
@@ -16,7 +16,7 @@ To begin, include the `BBCore` and `jQuery` libraries in your html (The latest v
 Then instantiate BBCore:
 
 ```javascript
-var bb = new BBCore({ accessToken: '<your api key>'});
+var bb = new BBCore({ accessToken: '<your access token or api key>', onerror: OnInvalidTokenCallback});
```
Comment (BenRussert) [inline README.md:19]: Was this part existing but just not documented before?

**Review by BenRussert — DISMISSED** [review]

File: README.md (hunk 3 above)
Comment (TomPendergrass) [reply README.md:19]: Yes

**Review by BenRussert — APPROVED** [review]

**Pull request merged**